adoptingerlang.org

```diff
@@ -1331,6 +1331,364 @@ Repeat this exercise from time to time, and eventually do some [[https://princip
+
+The problem would be that every time you want to try and modify a dependency to do something that its parent application needs, you'll have to commit and publish the change somewhere rebar3 can fetch it. This becomes a huge pain in the butt really fast.
+
+So there's a little trickshot of a feature called _checkout dependencies_. Checkout dependencies work as follows:
+
+- you have your main project somewhere on your file system
+- the dependency is declared in the rebar.config file of the main project
+- you also have that dependency somewhere on your file system, as a standalone project
+- you add a =_checkouts/= directory to the main project
+- you either copy or symlink the dependency's directory in the =_checkouts/= directory
+
+From that point on, every time the main app is built, it will add an =ebin/= directory to the dependency's directory in checkouts, and re-compile it as if it were a top-level application within the main app.
+
+You can then test the changes to your dependency within the main application until they're ready. Once you're done, remove the =ebin/= directory from the dependency, commit and publish your code in the dependency, remove it from the =_checkouts= directory, and =rebar3 upgrade= it.
+
+This lets you do a lot of small iterative changes locally on a dependency, within the context of the main app, without having to make everything public all the time. It massively reduces the overhead of per-app repositories as an overall development strategy.
+
+*** Using Elixir Dependencies
+
+For many years, the Elixir and Erlang road was a one way street. You could call Erlang from Elixir, but the opposite wasn't true. Since then, and thanks to the support of the Erlang Ecosystem Foundation, changes have been made to Rebar3 to give it a brand new compiler management structure. This structure has made it possible to write compiler plugins that can let Erlang users use Elixir code.
+
+The way to do it is to first install Elixir. For this you might want to follow the steps at LOCATION X. Most Elixir developers use =asdf= as a tool to manage versions. As described in TODO SECTION OF BUILD TOOLS, =kerl= options for Erlang can be used with the Erlang plugin for =asdf=, so that can give you a complete setup.
+
+With Elixir in place, add the following plugin to your library or project:
+
+TODO
+
+From that point on, you will be able to install any hex dependency containing Elixir code without a problem.
+
+TODO: check if we can use git deps now?
+
+Do note that mixed projects using both Erlang and Elixir within the same library are not currently supported since more work both on the rebar3 and Elixir side would need to be done to make this possible.
```

Comment:
```suggestion
Do note that mixed rebar3 projects using both Erlang and Elixir within the same library are not currently supported since more work both on the rebar3 and Elixir side would need to be done to make this possible.
```

Since this works in mix it is worth making it clear this is about rebar3 and not the languages general, except for the Elixir compiler issues and their not being available as a public api.